docs/pages/textarea.js

```diff
@@ -53,6 +53,265 @@ export default function DocsPage({ generatedDocGen }: {| generatedDocGen: DocGen
+          />
+        </MainSection.Subsection>
+
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="sm"
+            type="do"
+            description="Set the height of TextArea using row to ensure that the typical amount of text entered will be visible without needing to scroll."
+            defaultCode={`
+function Example(props) {
+  const [value, setValue] = React.useState('');
+
+  return (
+    <Box width="100%">
+      <TextArea
+        id="best-practices-do-height"
+        label="Have feedback on this product?"
+        onChange={({ value }) => {
+          setValue(value);
+        }}
+        placeholder="Tell us about your experience, what you love, or what we could improve."
+        helperText="Please don't submit passwords, email addresses, or other sensitive or personal info."
+        rows={5}
+        value={value}
+      />
+    </Box>
+  );
+}
+            `}
+          />
+          <MainSection.Card
+            cardSize="sm"
+            type="don't"
+            description="Set the row prop to less than 2. Use TextField instead, if only a single line of text is expected."
+            defaultCode={`
+function Example(props) {
+  const [value, setValue] = React.useState('');
+
+  return (
+    <Box width="100%">
+      <TextArea
+        id="best-practices-dont-row-height-1"
+        label="Send a message"
+        onChange={({ value }) => {
+          setValue(value);
+        }}
+        rows={1}
+        value={value}
+      />
+    </Box>
+  );
+}
+            `}
+          />
+        </MainSection.Subsection>
+      </MainSection>
+
+      <MainSection name="Accessibility">
+        <MainSection.Subsection
+          title="Comprehension"
+          description="Be sure to [provide instructions](https://www.w3.org/WAI/tutorials/forms/instructions/) to help users understand how to complete the form and use individual form controls."
+        />
+        <MainSection.Subsection
+          title="Labels"
+          description="Ensure the labels are precise and concise. Labels should only describe the text field they are associated with, and they must be visible. If you cannot use the `label` prop, ensure the alternative label's `htmlFor` attribute matches the TextArea's id. Labels are properly associated when clicking the label focuses the TextArea."
+        />
+        <MainSection.Subsection
+          title="Validation"
+          description={`
+    When providing a validation message, make sure the instructions are clear and help users complete the field. For example, "This field is required to submit". In addition, use the helper text to provide instructions to help users understand how to complete the text field or to indicate any needed input, allowed formats, timing limitations, or other pertinent information.
+    These practices give screen readers and users of assistive technologies more information about the form, helping them to fill it out.
```

Comment:
This sentence is a bit awkward: who does "them" refer to? It reads as both "screen readers" (objects) and "users of assistive technologies" (humans), which is weird. Since screen readers are an assistive technology, one option is to remove that reference, thus "them" refers only to people.

Reply:
I like it. Removed.